### DISCUSSION TITLE
Easier way to use Satori without JSX

### DISCUSSION THREAD

**Discussion by nightknighto** [vercel/satori#416] [Q&A] [answered]
I'm working on an express server that needs to use Satori. It is a node + express so no React or JSX there.

In the README it mentions using the react element object notation, but that can get very impractical for complex HTML structures.

I've also seen the satori-html package and tried using it, but it gives me this error:
![image](https://user-images.githubusercontent.com/52013393/222729613-1ef8fd1b-377b-4a28-950f-96695dcfc4a7.png)
If anyone knows the solution to this issue, that can be great. There are no mentions to this on their github and it doesn't seem to be active.

So now I'm looking for an easy way (maybe if there is an existing library?) to for example transform a string consisting of html structure like `"<div>Hello <span>World</span></div>` to the corresponding element objects to pass to Satori.

Any other solutions are welcomed as well!

**Comment by KiwiKilian** [comment]
`satori-html` is probably the best choice. You will either have to switch your express project to ESM or transpile `satori-html` to CommonJS to use it.

**Reply by nightknighto** [reply]
Could you give me directions to either? Especially that I'm working with TypeScript and so a lot of overhead when switching that I don't understand

**Comment by jeetiss** [comment, marked as answer]
This is an example of how to use ESM package `satory-html` in commonjs:

https://stackblitz.com/edit/node-cleajz?file=index.js

**Reply by nightknighto** [reply]
It didn't work unforunately, still gives me the same error.

Code for reference: https://github.com/Deadreyo/satory-experiment/pull/1/files

**Reply by nightknighto** [reply]
Solution works!
For full context, if using TypeScript then `moduleResolution` in `tsconfig` should be on `node16`.
